components/site-search/element.js

```diff
@@ -327,14 +334,17 @@ export class MDNSiteSearch extends L10nMixin(LitElement) {
                 })}</p>
               <ul class="site-search-results">
                 ${results.documents.map(
-                  (result) =>
+                  (result, index) =>
                     html`<li class="site-search-results__item">
                       <article>
                         <p class="site-search-results__path">
                           ${mdnUrl2Breadcrumb(result.mdn_url, this.locale)}
                         </p>
                         <h2 class="site-search-results__title">
-                          <a href=${result.mdn_url}>
+                          <a
+                            href=${result.mdn_url}
+                            data-glean-id=${`site-search: click -> ${1 + (results.metadata.page - 1) * results.metadata.size + index}`}
```

Comment:
Should we be gathering this for quicksearch too?

Reply:
I can imagine a future where we want to compare our simple title-based search in quicksearch with something more advanced, but we can always set up metrics at the time and A/B test or something.

Reply:
Found a mostly backward-compatible way of capturing the index for quick search and site search in a similar way, and added the url as well, as pages most frequently navigated to via page would be candidates for exposure in the menu.

See 4e1aa081a221c24ca3a39d2bdc28b2aa483b2c85.